sdks/python/apache_beam/transforms/core.py

```diff
@@ -3158,33 +3158,56 @@ def process(self, element):
           yield pvalue.TaggedOutput('hot', ((self._nonce % fanout, key), value))
 
     class PreCombineFn(CombineFn):
+      def __init__(self):
+        # Deepcopy of the combine_fn to avoid sharing state between lifted
+        # stages when using cloudpickle.
+        try:
+          self._combine_fn_copy = copy.deepcopy(combine_fn)
```

Comment:
If we:

1) replace this line with
          self._combine_fn_copy = pickler.loads(pickler.dumps(combine_fn))

2) use cloudpickle pickler

will we still hit  [https://github.com/apache/beam/issues/26209](https://www.google.com/url?sa=D&q=https%3A%2F%2Fgithub.com%2Fapache%2Fbeam%2Fissues%2F26209) ?

Reply:
>         except TypeError as e:

Can there be other errors? Let's catch all exceptions.

>               'Failed to copy combine function. Ensure python dependencies are'
              ' properly set up: %s'

I thought whether we should surface this, and I think this might create unactionable noise for the users. I wouldn't print anything if the copy, or copy via pickle was successful.